go/vt/vtgate/semantics/semantic_state.go

```diff
@@ -296,12 +296,26 @@ func vindexTableToColumnInfo(tbl *vindexes.Table) []ColumnInfo {
 	if tbl == nil {
 		return nil
 	}
+	nameMap := map[string]interface{}{}
 	cols := make([]ColumnInfo, 0, len(tbl.Columns))
 	for _, col := range tbl.Columns {
 		cols = append(cols, ColumnInfo{
 			Name: col.Name.String(),
 			Type: col.Type,
 		})
+		nameMap[col.Name.String()] = nil
+	}
+	for _, vindex := range tbl.ColumnVindexes {
+		for _, column := range vindex.Columns {
+			name := column.String()
+			if _, exists := nameMap[name]; exists {
+				continue
+			}
+			cols = append(cols, ColumnInfo{
+				Name: name,
+			})
+			nameMap[name] = nil
+		}
```

Comment:
optimization: this is not required if table is authoritative.

Reply:
ah, right. yeah, we could avoid reading the vindexes in that situation